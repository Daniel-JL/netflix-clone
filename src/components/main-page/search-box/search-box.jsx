import React, { useState } from 'react';
import styled from 'styled-components';
import { SearchButton } from '../../common/buttons/buttons';
import { SearchInput } from '../search-input/search-input';

const SearchBoxContainer = styled.div`
  display: flex;
  padding: 10px;
`;

const SearchBox = () => {
  const [searchActivated, setSearchActivated] = useState(false);

  return (
    <SearchBoxContainer>
      <SearchButton
        onClick={e => setSearchActivated(() => !searchActivated)}
      />
      {searchActivated && <SearchInput />}
    </SearchBoxContainer>
  );
};

export default SearchBox;