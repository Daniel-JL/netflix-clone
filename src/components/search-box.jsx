/* eslint-disable import/prefer-default-export */
import React, { useState } from 'react';
import styled from 'styled-components';
import { SearchButton } from './buttons';
import { SearchInput } from './search-input';

const SearchBoxContainer = styled.div`
  display: flex;
  padding: 10px;
`;

export const SearchBox = () => {
  const [searchActivated, setSearchActivated] = useState(false);

  return (
    <SearchBoxContainer>
      <SearchButton
        onClick={e => setSearchActivated(() => !searchActivated)}
      />
      {searchActivated && <SearchInput />}
    </SearchBoxContainer>
  );
};
