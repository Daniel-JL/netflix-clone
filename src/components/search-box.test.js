import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { SearchBox } from './search-box';

describe('Search box', () => {
  it('should render correctly', () => {
    const wrapper = renderer.create(<SearchBox />).toJSON();

    expect(wrapper).toMatchSnapshot();
  });

  it('should dismount text input if another view is activated', () => {

  });

  it('should dismount box if the magnifying glass symbol is pressed again', () => {

  });
});
