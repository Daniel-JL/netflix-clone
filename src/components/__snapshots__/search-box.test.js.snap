// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search box should render correctly 1`] = `
<div
  className="sc-fzoLsD hqjDOp"
>
  <button
    className="sc-AxgMl fUgHzy"
    onClick={[Function]}
  />
</div>
`;
